### PR TITLE
82 implement cypress

### PR DESCRIPTION
**Pull request by duaimei**
### What github issue is this PR for, if any?
Resolves #001

### What changed, and why?
Cypress adds a bunch of files :\, but now we'll have something to build off of!

### How will this affect user permissions?
no permission changes

### How is this tested? (please write tests!) 💖💪
This implements the cypress test suite nominally!

### Screenshots please :)
casa_cypress1

### CODE REVIEW

**Comment by duaimei** [comment]
I _thought_ that this was ready, but there are 50+ files that have been changed. I'm not 100% sure why there is a cypress folder under spec, and why that has good looking code.

I will clean up the files and then take off the WIP tag

**Comment by jmkoni** [comment]
@duaimei did you mean to downgrade a bunch of stuff in the gemfile.lock? I don't think we want to do that...

File: .travis.yml [hunk 1]
```diff
@@ -41,4 +41,4 @@ script:
   - RUBYOPT='-W:no-deprecated -W:no-experimental' bundle exec rspec
 
 after_script:
-  - ./cc-test-reporter after-build --exit-code $TRAVIS_TEST_RESULT
+  - ./cc-test-reporter after-build --exit-code $TRAVIS_TEST_RESULT
```
Comment (compwron) [inline .travis.yml:44]: Removing the line ending isn't a cleanup but linter will get it when we run that again after hacktober :)

File: CYPRESS.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+running cypress locally involves two terminal windows:
```
Comment (compwron) [inline CYPRESS.md:1]: yay docs!

**Review by compwron — APPROVED** [review]
Very cool! I'm going to merge and then re-upgrade gems.

**Pull request merged**